internal/artifact/artifact.go

```diff
@@ -0,0 +1,71 @@
+package artifact
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+	"github.com/ryanuber/go-glob"
+	"github.com/saucelabs/saucectl/internal/config"
+	"github.com/saucelabs/saucectl/internal/job"
+)
+
+// Downloader defines artifacts downloader
+type Downloader struct {
+	JobReader job.Reader
+}
+
+// DownloadArtifacts downloads artifacts according to config
+func (d *Downloader) DownloadArtifacts(artifactsCfg config.ArtifactDownload, jobID string, passed bool) {
```

Comment:
Should `artifactsCfg config.ArtifactDownload` maybe be a member variable of `Downloader`? I don't see the usecase of it ever changing between invocations of `DownloadArtifacts`. Seems to be more of a configuration of the artifact downloader (which it is :D). That'd also make the signature of the other methods cleaner, as they don't have to pass around the artifact config object.

Reply:
make sense! Updating